rust/src/logger.rs

```diff
@@ -0,0 +1,152 @@
+//! This module implements logging facilities.
+//!
+//! # FFI
+//!
+//! [`stracciatella::c::logger`] contains a C interface for this module.
+//!
+//! [`stracciatella::c::logger`]: ../c/logger/index.html
+
+use lazy_static::lazy_static;
+use log::*;
+use simplelog::*;
+use std::default::Default;
+use std::fs::File;
+use std::path::Path;
+use std::sync::atomic::Ordering;
+use std::sync::{atomic, Arc};
+
+lazy_static! {
+    /// Global log level used to filter messages at runtime
+    static ref GLOBAL_LOG_LEVEL: Arc<atomic::AtomicUsize> = {
```

Comment:
You're using AtomicUsize so you don't need `lazy_static` or 'Arc`.
See "Atomic Access" in https://docs.rust-embedded.org/book/concurrency/

Reply:
True, I used this so I can call `LogLevel::default`, but I think just initializing with a specific value makes it clear which log level is default as well...